toolchain/check/cpp/location.cpp

```diff
@@ -9,17 +9,49 @@
 
 namespace Carbon::Check {
 
-static auto GetFile(Context& context, SemIR::CheckIRId ir_id)
-    -> const SemIR::File* {
-  if (ir_id == context.sem_ir().check_ir_id()) {
-    // Common case: the IR is the current file.
-    return &context.sem_ir();
-  }
+namespace {
+struct FileInfo {
+  const SemIR::File* sem_ir;
+  clang::SourceLocation start_loc;
+};
+}  // namespace
+
+// Map a CheckIRId into information about the corresponding file in both SemIR
+// and Clang's source manager.
+static auto GetFileInfo(Context& context, SemIR::CheckIRId ir_id) -> FileInfo {
+  const SemIR::File* sem_ir = &context.sem_ir();
+  unsigned file_index = 0;
```

Comment:
Wouldn't style say `int` here, avoiding `unsigned`?

```suggestion
  int file_index = 0;
```

[Style here](https://google.github.io/styleguide/cppguide.html#Integer_Types), I think current advice is specifically `int` or a stdint.h type -- so also if you're specifically trying for unsigned handling, it might be good to document why and use `uint32_t` or `int64_t`.

```suggestion
  // Use 64-bit in case we have have int max files open.
  int64_t file_index = 0;
```

```suggestion
  // Use unsigned in case we have have int max files open.
  uint32_t file_index = 0;
```

(but can this case happen?)